apps/backend/src/libs/modules/open-ai/open-ai.module.ts

```diff
@@ -0,0 +1,273 @@
+import { OpenAI } from "openai";
+import { zodResponseFormat } from "openai/helpers/zod";
+
+import { type Config } from "~/libs/modules/config/config.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Logger } from "~/libs/modules/logger/logger.js";
+
+import {
+	OpenAiAssistantConfig,
+	OpenAIErrorMessage,
+} from "./libs/enums/enums.js";
+import { OpenAIError } from "./libs/exceptions/exceptions.js";
+import {
+	type OpenAiRequestMessage,
+	type OpenAiResponseMessage,
+	type OpenAiRunThreadRequestDto,
+} from "./libs/types/types.js";
+
+type Constructor = {
+	config: Config;
+	logger: Logger;
+};
+
+class OpenAi {
+	private assistantId: null | string = null;
+	private config: Config;
+	private logger: Logger;
+	private openAi: OpenAI;
+
+	public constructor({ config, logger }: Constructor) {
+		this.config = config;
+		this.logger = logger;
+		this.openAi = this.createOpenAi();
+	}
+
+	private createOpenAi(): OpenAI {
+		return new OpenAI({
+			apiKey: this.config.ENV.OPEN_AI.API_KEY,
+		});
+	}
+
+	private async getOrInitializeAssistant(): Promise<string> {
+		try {
+			const existingAssistants = await this.openAi.beta.assistants.list();
+			const assistant = existingAssistants.data.find(
+				(assistant) => assistant.name === OpenAiAssistantConfig.NAME,
+			);
+
+			const initializedAssistant =
+				assistant ??
+				(await this.openAi.beta.assistants.create({
+					instructions: OpenAiAssistantConfig.INSTRUCTION,
+					model: this.config.ENV.OPEN_AI.MODEL,
+					name: OpenAiAssistantConfig.NAME,
+					temperature: OpenAiAssistantConfig.TEMPERATURE,
+					tools: [...OpenAiAssistantConfig.TOOLS],
+					top_p: OpenAiAssistantConfig.TOP_P,
+				}));
+
+			this.logger.info(
+				`AI Assistant "${initializedAssistant.name as string}" is fully ready for interaction.`,
+			);
+			this.logger.info(`AI Assistant model - "${initializedAssistant.model}".`);
+
+			return initializedAssistant.id;
+		} catch (error) {
+			this.logger.error(`Error initializing AI assistant: ${String(error)}`);
+
+			throw new OpenAIError({
+				message: OpenAIErrorMessage.WRONG_RESPONSE,
+				status: HTTPCode.INTERNAL_SERVER_ERROR,
+			});
+		}
+	}
+
+	private async handleRequiresAction(
+		run: OpenAI.Beta.Threads.Runs.Run,
+		functionName: string,
+	): Promise<void> {
+		if (!run.required_action) {
+			return;
+		}
+
+		const { tool_calls } = run.required_action.submit_tool_outputs;
+
+		await Promise.all(
+			tool_calls.map(
+				async ({
+					function: toolFunction,
+					function: { arguments: arguments_ },
```

Comment:
```suggestion
					function: { arguments: arguments },
```

Reply:
Can't use `arguments `
TS1210: Code in strict mode does not allow this use of arguments